udata_front/theme/gouvfr/assets/js/pages/Admin/Profile/Profile.vue

```diff
@@ -0,0 +1,168 @@
+                          <p>{{ t("If you want to delete your published content too, start by deleting the contents before deleting your account.") }}</p>
+                        </div>
+                        <div class="fr-modal__footer">
+                          <div class="fr-btns-group fr-btns-group--right fr-btns-group--inline-reverse fr-btns-group--inline-lg fr-btns-group--icon-left">
+                              <button
+                                class="fr-btn fr-btn--secondary fr-btn--secondary--error"
+                                role="button"
+                                :disabled="loading"
+                                @click="deleteCurrentOrganization"
+                              >
+                                  {{ t("Delete") }}
+                              </button>
+                          </div>
+                      </div>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </dialog>
+            </Teleport>
+          </div>
+        </AdminDangerZone>
+      </template>
+    </DescribeOrganizationFrom>
+  </div>
+</template>
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { useI18n } from "vue-i18n";
+import Breadcrumb from "../../../components/Breadcrumb/Breadcrumb.vue";
+import AdminDangerZone from "../../../components/AdminDangerZone/AdminDangerZone.vue";
+import DescribeOrganizationFrom from "../../OrganizationPublishingForm/Step2DescribeOrganization.vue";
+import type { Me, OrganizationV1 } from "../../../types";
+import { fetchMe } from "../../../api/me";
+import { watchEffect } from 'vue';
+import { deleteOrganization, getOrganization, updateOrganization } from '../../../api/organizations';
+import { useToast } from '../../../composables/useToast';
+import { getRandomId } from '@etalab/data.gouv.fr-components';
+import { useRouter } from 'vue-router';
+
+const { t } = useI18n();
+const { toast } = useToast();
+const props = defineProps<{oid: string}>();
+const router = useRouter();
+const me = ref<Me | null>(null);
+const form = ref<InstanceType<typeof DescribeOrganizationFrom> | null>(null);
+
+// TODO : use `useCurrentOrganization` when merged
+const organization = ref<OrganizationV1 | null>(null);
+const errors = ref([]);
+const loading = ref(false);
+const modalId = getRandomId("modal");
+const modalTitleId = getRandomId("modalTitle");
+
+function deleteCurrentOrganization() {
+  if(organization.value) {
+    loading.value = true;
+    deleteOrganization(organization.value.id)
+    .then(() => {
+      router.replace("/");
+      window.location.reload();
+    })
+    .catch(() => toast.error(t("An error occured when deleting the organization.")))
+    .finally(() => loading.value = false);
+  }
```

Comment:
You could early return here but it's a matter of preference :-)

```suggestion
  if(!organization.value) {
    return;
  }
  loading.value = true;
  deleteOrganization(organization.value.id)
    .then(() => {
      router.replace("/");
      window.location.reload();
    })
    .catch(() => toast.error(t("An error occured when deleting the organization.")))
    .finally(() => loading.value = false);
  }
```

Reply:
I tend to use `if` to wrap code when I'm doing it only for the TS type-checker instead of an early exit because the `!organization.value`  can't really exist (the form is hidden without org).

Does this make sense to you ?